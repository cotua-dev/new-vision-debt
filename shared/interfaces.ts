export interface PageProps {
    props: {};
    revalidate: number;
};
