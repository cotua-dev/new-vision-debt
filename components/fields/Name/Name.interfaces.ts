import { Dispatch, SetStateAction } from 'react';
import { Step } from '../../Stepper/Stepper.interfaces';

export interface NameProps {
    steps: Step[];
    currentStep: number;
    setDisableNextButton?: Dispatch<SetStateAction<boolean>>;
    firstName: string;
    lastName: string;
    setFirstName: Dispatch<SetStateAction<string>>;
    setLastName: Dispatch<SetStateAction<string>>;
};
