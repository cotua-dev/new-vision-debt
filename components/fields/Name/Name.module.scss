@use 'styles/variables';
@use 'styles/mixins';

div.name-fields-wrapper {
    @include mixins.flexify(center, false, column);

    div.name-field-wrapper {
        @include mixins.flexify(flex-start, false, column);
        margin: 20px 0;

        label.name-field-label {
            color: variables.$primary-color;
            font-size: 18px;
            font-weight: 700;
        }

        input.name-field {
            @include mixins.base-field;
            margin: 0;
        }
    }
}
