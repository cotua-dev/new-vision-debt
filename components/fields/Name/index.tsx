import { ChangeEvent } from 'react';
import { NameProps } from './Name.interfaces';
import styles from './Name.module.scss';

export function Name(props: NameProps): JSX.Element {
    const { steps, currentStep, setDisableNextButton, firstName, lastName, setFirstName, setLastName } = props;

    /**
     * Set the value of first name for the step and `firstName`
     * @param value String of the first name
     */
    function handleFirstNameSetting(value: string): void {
        // Set the `firstName`
        setFirstName(value);

        // Set the current step's `firstName` value
        steps[currentStep].value.firstName = value;
    }

    /**
     * Set the value of last name for the step and `lastName`
     * @param value String of the last name
     */
    function handleLastNameSetting(value: string): void {
        // Set the `lastName`
        setLastName(value);

        // Set the current step's `lastName` value
        steps[currentStep].value.lastName = value;

        // Make the current step valid
        steps[currentStep].validity = true;

        // Enable the next button
        setDisableNextButton && setDisableNextButton(false);
    }

    return (
        <div className={styles['name-fields-wrapper']}>
            <div className={styles['name-field-wrapper']}>
                <label
                    className={styles['name-field-label']}
                    htmlFor="firstName"
                >{`First Name`}</label>
                <input
                    className={styles['name-field']}
                    type="text"
                    id="firstName"
                    name="firstName"
                    onChange={(e: ChangeEvent<HTMLInputElement>) => handleFirstNameSetting(e.target.value)}
                    value={firstName}
                />
            </div>
            <div className={styles['name-field-wrapper']}>
                <label
                    className={styles['name-field-label']}
                    htmlFor="lastName"
                >{`Last Name`}</label>
                <input
                    className={styles['name-field']}
                    type="text"
                    id="lastName"
                    name="lastName"
                    onChange={(e: ChangeEvent<HTMLInputElement>) => handleLastNameSetting(e.target.value)}
                    value={lastName}
                />
            </div>
        </div>
    );
}
