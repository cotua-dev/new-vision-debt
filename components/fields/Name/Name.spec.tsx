import { render, RenderResult } from '@testing-library/react';
import { Name } from './index';
import { NameProps } from './Name.interfaces';

describe('Name Component', (): void => {
    let renderResult: RenderResult;
    const props: NameProps = {
        steps: [],
        currentStep: 0,
        firstName: '',
        lastName: '',
        setFirstName: () => {},
        setLastName: () => {},
    };

    beforeAll((): RenderResult => renderResult = render(<Name {...props}/>));

    test('renders', (): void => {
        const { container } = renderResult;
        expect(container.firstChild).toMatchSnapshot();
    });

    afterAll((): void => renderResult.unmount());
});
