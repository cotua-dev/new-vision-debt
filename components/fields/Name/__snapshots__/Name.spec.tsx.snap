// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name Component renders 1`] = `
<div
  class="name-fields-wrapper"
>
  <div
    class="name-field-wrapper"
  >
    <label
      class="name-field-label"
      for="firstName"
    >
      First Name
    </label>
    <input
      class="name-field"
      id="firstName"
      name="firstName"
      type="text"
      value=""
    />
  </div>
  <div
    class="name-field-wrapper"
  >
    <label
      class="name-field-label"
      for="lastName"
    >
      Last Name
    </label>
    <input
      class="name-field"
      id="lastName"
      name="lastName"
      type="text"
      value=""
    />
  </div>
</div>
`;
